zebra/kernel_netlink.c

```diff
@@ -188,7 +188,10 @@ static int netlink_socket(struct nlsock *nl, unsigned long groups,
 		return -1;
 	}
 
-	sock = socket(AF_NETLINK, SOCK_RAW, NETLINK_ROUTE);
+	if (ns_id != NS_DEFAULT)
```

Comment:
I thought the whole point of the wrappered socket call was to not have to do this type of if statement.  ns_socket should just do the *right* thing, why is it not?

Reply:
this will be simplified, like it has been done in change just below.

Reply:
done